internal/impl/pkgs.go

```diff
@@ -254,3 +259,46 @@ type Package struct {
 	// package depends on. It does not contain transitive dependencies.
 	DirectDependencies []Package
 }
+
+func (d *Devbox) pendingPackagesForInstallation() ([]string, error) {
+	pkgs := []string{}
+	profileDir, err := d.profileDir()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	cmd := exec.Command("nix-env", "-q", "-p", profileDir, "--json")
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	installed := map[string]struct {
+		Name string `json:"name"`
+	}{}
+	if err := json.Unmarshal(out, &installed); err != nil {
+		return nil, errors.WithStack(err)
+	}
+	for _, i := range installed {
+		// rekey by name
+		installed[i.Name] = i
+	}
+
+	for _, pkg := range d.cfg.Packages {
+		info, found := nix.PkgInfo(d.cfg.Nixpkgs.Commit, pkg)
```

Comment:
Unfortunately, this line will hurt perf. I had it in my original code as well, and needed to remove it to make the new UX perf-neutral. Because it relies on flakes-oriented nix apis.

The impact was pretty significant for the go.1.19 example project:
- local `devbox shell` went from 5 seconds to 8 seconds.
- `devbox cloud shell` went from 37 seconds to 55 seconds

Reply:
@savil FYI, this is fixed. `nix.PkgInfo` can now do flakes or no flakes so it no longer incurs the flake download penalty if the flag is turned off. The non-flakes version will use the same version of nixpkgs as `nix-instantiate`.